### PR TITLE
Derive some traits for ed25519::{PublicKey, Signature}

### PR DESCRIPTION
**Pull request by thibault-martinez**
# Description of change

Derive some common traits for public types.
Without the traits, these types are basically unusable by bee.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by arnauorriols** [comment]
What's the status of this PR? IOTA-Streams would be interested in it getting merged 🙂

**Comment by thibault-martinez** [comment]
Last time I checked, some trait impls were missing in the dependencies and the owners didn't reply, I'll try to check again

**Comment by arnauorriols** [comment]
We could separate the serde traits in a different PR, and get the basic ones merged.

File: src/signatures/ed25519.rs [hunk 1]
```diff
@@ -35,6 +35,8 @@ impl SecretKey {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
```
Comment (arnauorriols) [inline src/signatures/ed25519.rs:38]: Why not derive `Hash`, `Display`, `PartialEq` and `Eq` for `PublicKey`?

**Pull request closed without merging**